Sources/Articles/Documentation.docc/2025/jwt-kit/jwt-kit.md

```diff
@@ -0,0 +1,198 @@
+# Getting Started with Vapor's JWTKit 5: A Practical Introduction to JSON Web Tokens in Swift
+
+JWTKit is a library for working with JSON Web Tokens (JWT) in Swift. It provides a simple and easy-to-use API for creating, parsing, and validating JWTs. JWTs are a popular way to securely transmit information between parties, and JWTKit makes it easy to work with them in your Swift projects.
+
+In this tutorial, you'll learn how to use JWTKit to create and validate JWTs in Swift. You'll start by installing JWTKit using Swift Package Manager, then you'll learn how to create a JWT and validate it using JWTKit's API. At the end of this tutorial, you'll be able to implement an authentication and authorisation flow in your Swift projects using JWTs.
+
+## What is a JSON Web Token (JWT)?
+
+A JWT is a compact, URL-safe way to represent claims between two parties. The token is digitally signed, which makes it secure and trustworthy. JWTs are commonly used for authentication and authorisation in web applications, APIs, and microservices.
+
+A JWT consists of three parts: a header, a payload, and a signature. The header contains metadata about the token, such as the type of token and the algorithm used to sign it. The payload contains the claims, which are statements about an entity (typically the user) and additional data. The signature is used to verify that the token is valid and has not been tampered with.
+
+It usually looks like this:
+```
+eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c
+```
+
+If you inspect the token on [jwt.io](https://jwt.io), you'll see that it consists of three parts separated by dots: the header, the payload, and the signature.
+
+> Note: don't paste production tokens into jwt.io or any other online tool, as they could be intercepted and misused.
+
+## Installing JWTKit
+
+JWTKit is available as a Swift package, which makes it easy to install and use in your projects. To install it, add it as a dependency in your `Package.swift` file:
+
+```swift
+dependencies: [
+    .package(url: "https://github.com/vapor/jwt-kit.git", from: "5.0.0")
+]
+```
+
+Then, add JWTKit as a dependency to your target:
+
+```swift
+targets: [
+    .target(name: "YourTarget", dependencies: [
+        .product(name: "JWTKit", package: "jwt-kit")
+    ])
+]
+```
+
+## Configuration
+
+In JWTKit everything revolves around the ``JWTKeyCollection`` object. As the name suggests, it's a collection of keys that can be used to sign and verify JWTs. The declaration is simple, you can add this in your app's configuration code:
+
+<!-- @Snippet(path: "site/Snippets/JWTKit", slice: key_collection_init) -->
+
+```swift
+let keyCollection = JWTKeyCollection()
+```
+
+If you know the signing algorithm you want to use, adding keys to the collection is also straightforward:
+
+<!-- @Snippet(path: "site/Snippets/JWTKit", slice: key_collection_add_hmac) -->
```

Comment:
```suggestion
@Snippet(path: "site/Snippets/jwt-kit", slice: key_collection_add_hmac)
```

I believe your snippets didn't work because the filename isn't `JWTKit.swift` but `jwt-kit.swift` (note we need to omit the file extension)